src/controllers/entry.controller.ts

```diff
@@ -163,8 +172,55 @@ export class GoodEntryController {
       },
       description: '',
     })
-    entry: Omit<GoodEntry, 'id'>,
+    entry: Partial<GoodEntry>,
   ): Promise<unknown> {
-    return this.goodEntryRepository.create(entry);
+    console.log('Data received:', entry);
+    const pogProfileParams: PogProfileParams = {
+      userId: '',
+      email: '',
+      doGooder: '',
+    };
+
+    if (entry.userId) {
+      Object.assign(pogProfileParams, {userId: entry.userId});
+    }
+
+    if (entry.email) {
+      Object.assign(pogProfileParams, {email: entry.email});
+    }
+
+    if (entry.doGooder) {
+      Object.assign(pogProfileParams, {doGooder: entry.doGooder});
+    }
```

Comment:
could this be simplified?
```suggestion
    const pogProfileParams: PogProfileParams = {
      userId: entry?.userId ?? '',
      email: entry?.email ?? '',
      doGooder: entry?.doGooder ?? '',
    };
```